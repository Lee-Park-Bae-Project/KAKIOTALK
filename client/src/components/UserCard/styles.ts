import styled from 'styled-components';
import Flex from 'atoms/Flex';
import { color, fontSize } from 'styles/global';

export const Container = styled(Flex)`
  height: 5rem;
  justify-content: flex-start;
  padding: 0 1rem;
  background-color: ${color.WHITE};
  &:hover{
    background-color: ${color.HOVER_GRAY};
  }
`;

export const ProfileWrapper = styled(Flex)`
  justify-content: flex-start;

`;

export const StatusMessageWrapper = styled.span`
  font-size: ${fontSize.SMALL};
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;
