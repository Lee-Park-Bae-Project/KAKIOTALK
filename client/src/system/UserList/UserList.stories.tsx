import React from 'react';
import UserList from 'system/UserList';
import shortid from 'shortid';
import faker from 'faker';

export default {
  title: 'Systems/UserList',
  component: UserList,
};

const myProfile = {
  uuid: shortid.generate(),
  email:faker.internet.email(),
  name: faker.internet.userName(),
  statusMessage: faker.lorem.lines(),
};

const userList = [
  {
    uuid: shortid.generate(),
    name: faker.internet.userName(),
    email:faker.internet.email(),
    statusMessage: faker.lorem.lines(),
  },
  {
    uuid: shortid.generate(),
    name: faker.internet.userName(),
    email:faker.internet.email(),
    statusMessage: faker.lorem.lines(),
  },
];

export const DefaultUserCard = () => {
  const onClick = (id: string) => () => {
    alert(id);
  };
  return (
    <UserList
      myProfile={myProfile}
      userList={userList}
      onClick={onClick}
    />
  );
};
