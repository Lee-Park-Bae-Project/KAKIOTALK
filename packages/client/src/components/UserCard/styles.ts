import styled from 'styled-components'
import Flex from 'atoms/Flex'
import {
  color, fontSize,
} from 'styles/global'

export const Container = styled(Flex)`
  height: 5rem;
  justify-content: flex-start;
  padding: 0 1rem;
  background-color: ${color.WHITE};
  &:hover {
    background-color: ${color.HOVER_GRAY};
  }
`

export const ProfileWrapper = styled(Flex)`
  justify-content: flex-start;
`

export const StatusMessageWrapper = styled.span`
  font-size: ${fontSize.SMALL};
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`
export const Tri = styled.div`
  width: 0px;
  height: 0px;
  border-top: 15px solid transparent;
  border-right: 30px solid gray;
  border-bottom: 15px solid transparent;
`
