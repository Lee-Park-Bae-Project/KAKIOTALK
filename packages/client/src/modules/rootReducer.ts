import { combineReducers } from 'redux'
import { persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import friends from 'modules/friends'
import profile from 'modules/profile'
import login from 'modules/login'
import room from 'modules/room'
import chat from 'modules/chat'
import alert from 'modules/alert'

const persistConfig = {
  key: 'root',
  storage,
  blacklist: [],

}
const rootReducer = combineReducers({
  friends,
  profile,
  login,
  room,
  chat,
  alert,
})

export default persistReducer(persistConfig, rootReducer)
