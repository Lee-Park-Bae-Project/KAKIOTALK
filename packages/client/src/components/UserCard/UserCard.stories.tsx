import React, { FC } from 'react'
import UserCard from './index'

export default {
  title: 'Component/UserCard',
  component: UserCard,
}

export const BasicUserCard: FC = () => (
  <UserCard
    name='junow'
    statusMessage='Brand New 2020'
    />
)

export const UserCardWithClickHandler: FC = () => {
  const name = 'junow'
  const onClick = () => console.warn(name)
  return (
    <UserCard
      name={name}
      onClick={onClick}
      statusMessage='Brand New 2020'
    />
  )
}

export const UserCardWithLongText = () => (
  <UserCard
    name='junow'
    statusMessage='was it a cat i saw was it a cat i saw was it a cat i saw'
  />
)

UserCardWithLongText.story = { name: '긴 텍스트가 들어있는 UserCard' }
