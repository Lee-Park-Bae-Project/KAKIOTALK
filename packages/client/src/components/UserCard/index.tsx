import React, {
  FC, useCallback, useState,
} from 'react'
import * as S from 'components/UserCard/styles'
import { color } from 'styles/global'
import TextIcon from 'components/TextIcon'
import Profile from 'system/Profile'
import { useOutsideClick } from 'common/utils'

interface UserCardProp {
  /** user uuid */
  uuid?: string
  /** user card 에 표시할 유저 네임 */
  name: string
  /** 유저 상태메시지 */
  statusMessage?: string
  /** 클릭핸들러 */
  onClick?: () => void
  /** 유저 프로필사진 URL */
  imageUrl?: string
  /** 본인 or 친구 중 누구의 프로필인지 판별 */
  isMyProfile: boolean
}

/**
 * UserCard
 *
 * - 유저의 프로필 사진과 이름이 표시됩니다.
 */
const UserCard: FC<UserCardProp> = ({
  uuid = '',
  name,
  statusMessage = '',
  imageUrl,
  isMyProfile,
  onClick = undefined,
}) => {
  const [isClicked, setIsClicked] = useState(false)
  const [isOverflow, setIsOverflow] = useState(false)
  const handlePopUpClick = () => {
    setIsClicked(!isClicked)
  }
  const profileRef = React.useRef(null)
  useOutsideClick(profileRef, () => {
    if (isClicked) setIsClicked(false)
  })
  const setOverflow = useCallback((node: HTMLDivElement) => {
    if (node && node.getBoundingClientRect().bottom < 300) {
      setIsOverflow(true)
    }
  }, [])
  return (
    <S.Container ref={setOverflow} onClick={onClick}>
      <S.ProfileWrapper>
        <TextIcon
          icon='Account'
          color={color.GRAY}
          text={name}
          onClick={handlePopUpClick}
          imageUrl={imageUrl}
          statusMessage={statusMessage}
          textSize="15px"
        >
          {isClicked && (
            <React.Fragment>
            <S.Tri/>
            <Profile
              uuid={uuid}
              name={name}
              statusMessage={statusMessage}
              handleCloseClick={handlePopUpClick}
              imageUrl={imageUrl || ''}
              profileRef={profileRef}
              isMyProfile={isMyProfile}
              isOverflow={isOverflow}
            />
            </React.Fragment>
          )}
        </TextIcon>

      </S.ProfileWrapper>
    </S.Container>
  )
}

export default UserCard
