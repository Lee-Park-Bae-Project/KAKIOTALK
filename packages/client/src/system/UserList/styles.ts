import styled from 'styled-components';
import Flex from 'atoms/Flex';
import { color } from 'styles/global';

export const Container = styled(Flex)`
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  margin-top: 1rem;
  margin-bottom: auto;
  background-color: ${color.WHITE};
  /* height: 100vh; */
  
  width: 100%;
  margin: 0;
  margin-right: auto;
  margin-bottom: auto;
`;

export const Friend = styled(Flex)`
  flex-direction: row;
  justify-content: flex-start;
  padding: 0 1rem;
`;

export const H3 = styled.h3`
  text-indent: 1rem;
`;
export const P = styled.span`
  text-indent: 0.5rem;
`;
