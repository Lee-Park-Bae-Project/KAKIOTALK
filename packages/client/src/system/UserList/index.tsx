import React, { FC } from 'react'
import * as S from 'system/UserList/styles'
import UserCard from 'components/UserCard'
import Hr from 'atoms/Hr'
import { Models } from '@kakio/common'

interface Props{
  /** 내 정보 */
  myProfile: Models.User;
  /** 친구 목록을 담은 배열 */
  userList: Models.User[];
  /** 각 친구들 클릭 했을 때 클릭 이벤트 핸들러 */
  onClick: (uuid: string) => () => void;
}
/**
 * - UserList 를 나열하는 컴포넌트
 * - ex) 친구목록
 */
const UserList: FC<Props> = ({
  myProfile,
  userList,
  onClick,
}) => (
    <S.Container>
      <UserCard
        name={myProfile.name}
        isMyProfile={true}
        />
      <Hr/>
      <S.Friend>
        <S.H3>친구</S.H3>
        <S.P>{userList.length}</S.P>
      </S.Friend>
      {
        userList.map((user) => (
          <UserCard
            key={user.uuid}
            name={user.name}
            statusMessage={user.statusMessage}
            onClick={onClick(user.uuid)}
            isMyProfile={false}
          />
        ))
      }
    </S.Container>
)

export default UserList
