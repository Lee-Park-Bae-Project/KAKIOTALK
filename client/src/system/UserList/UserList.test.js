/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import UserList from 'system/UserList';
import UserCard from 'components/UserCard';

import {
  addFriends,
} from 'modules/friends';
import rootReducer from 'modules';
import { createStore } from 'redux';

const store = createStore(rootReducer);
const myProfile = {
  id: '123',
  userName: 'junow',
  statusMessage: 'im junow',
};
const userList = [
  {
    id: '234',
    userName: 'park',
    statusMessage: 'parkpark',
  },
  {
    id: '345',
    userName: 'bae',
    statusMessage: 'baebae',
  },
];

describe('<UserList/>', () => {
  it('match snapshot', () => {
    const onClick = (id) => () => { console.log(id); };
    const wrapper = mount(
    <UserList
      myProfile={myProfile}
      userList={userList}
      onClick={onClick}
    />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('matches props', () => {
    const onClick = (id) => () => { console.log(id); };
    const wrapper = mount(
      <UserList
        myProfile={myProfile}
        userList={userList}
        onClick={onClick}
      />
    );

    expect(wrapper.props().myProfile).toEqual(myProfile);
    expect(wrapper.props().userList).toEqual(userList);
  });

  it('click handler', () => {
    let result = '';
    const onClick = (id) => () => {
      result = id;
    };

    const wrapper = mount(
      <UserList
        myProfile={myProfile}
        userList={userList}
        onClick={onClick}
      />
    );

    const firstUser = wrapper.find(UserCard).at(1);
    const secondUser = wrapper.find(UserCard).at(2);
    expect(result).toBe('');
    firstUser.simulate('click');
    expect(result).toBe(userList[0].id);
    secondUser.simulate('click');
    expect(result).toBe(userList[1].id);
  });
});
