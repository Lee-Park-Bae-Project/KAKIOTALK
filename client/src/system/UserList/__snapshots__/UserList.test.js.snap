// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList/> match snapshot 1`] = `
<UserList
  myProfile={
    Object {
      "id": "123",
      "statusMessage": "im junow",
      "userName": "junow",
    }
  }
  onClick={[Function]}
  userList={
    Array [
      Object {
        "id": "234",
        "statusMessage": "parkpark",
        "userName": "park",
      },
      Object {
        "id": "345",
        "statusMessage": "baebae",
        "userName": "bae",
      },
    ]
  }
>
  <Styled(styled.div)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "bQJAsI",
            "rules": Array [
              "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
              "
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  margin-top: 1rem;
  margin-bottom: auto;
  background-color: ",
              "#FEFFFF",
              ";
  /* height: 100vh; */
  
  width: 100%;
  margin: 0;
  margin-right: auto;
  margin-bottom: auto;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa sc-bwzfXH bQJAsI"
      >
        <UserCard
          userName="junow"
        >
          <Styled(styled.div)>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-EHOje",
                    "isStatic": false,
                    "lastClassName": "VYauM",
                    "rules": Array [
                      "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                      "
  height: 5rem;
  justify-content: flex-start;
  padding: 0 1rem;
  background-color: ",
                      "#FEFFFF",
                      ";
  &:hover{
    background-color: ",
                      "#F1F2F2",
                      ";
  }
",
                    ],
                  },
                  "displayName": "Styled(styled.div)",
                  "foldedComponentIds": Array [
                    "sc-bdVaJa",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-EHOje",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa sc-EHOje VYauM"
              >
                <Styled(styled.div)>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bZQynM",
                          "isStatic": false,
                          "lastClassName": "iEkADQ",
                          "rules": Array [
                            "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                            "
  justify-content: flex-start;

",
                          ],
                        },
                        "displayName": "Styled(styled.div)",
                        "foldedComponentIds": Array [
                          "sc-bdVaJa",
                        ],
                        "render": [Function],
                        "styledComponentId": "sc-bZQynM",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bdVaJa sc-bZQynM iEkADQ"
                    >
                      <TextIcon
                        color="#C6C3C1"
                        icon="Account"
                        text="junow"
                      >
                        <styled.div
                          iconPosition="left"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-htoDjs",
                                  "isStatic": false,
                                  "lastClassName": "bdJNWD",
                                  "rules": Array [
                                    "
    display: flex;
    flex-direction: ",
                                    [Function],
                                    ";
    width: fit-content;
    height: fit-content;
    align-items: center;
    justify-content: center;
  ",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-htoDjs",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            iconPosition="left"
                          >
                            <div
                              className="sc-htoDjs bdJNWD"
                            >
                              <Icon
                                color="#C6C3C1"
                                icon="Account"
                              >
                                <ForwardRef(SvgAccount)
                                  fill="#C6C3C1"
                                  height="2rem"
                                  width="2rem"
                                >
                                  <svg
                                    fill="#C6C3C1"
                                    height="2rem"
                                    width="2rem"
                                  >
                                    account.svg
                                  </svg>
                                </ForwardRef(SvgAccount)>
                              </Icon>
                              <styled.span
                                text="junow"
                                textColor="black"
                                textSize="1rem"
                              >
                                <StyledComponent
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-dnqmqq",
                                        "isStatic": false,
                                        "lastClassName": "eCurCS",
                                        "rules": Array [
                                          "
    color: ",
                                          [Function],
                                          ";
    line-height: 2rem;
    padding: 1rem;
  ",
                                        ],
                                      },
                                      "displayName": "styled.span",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "sc-dnqmqq",
                                      "target": "span",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  text="junow"
                                  textColor="black"
                                  textSize="1rem"
                                >
                                  <span
                                    className="sc-dnqmqq eCurCS"
                                  >
                                    junow
                                  </span>
                                </StyledComponent>
                              </styled.span>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </TextIcon>
                    </div>
                  </StyledComponent>
                </Styled(styled.div)>
              </div>
            </StyledComponent>
          </Styled(styled.div)>
        </UserCard>
        <styled.hr>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-iwsKbI",
                  "isStatic": false,
                  "lastClassName": "byAdSv",
                  "rules": Array [
                    "
  width: 90%;
  margin: 1rem auto;
  border-color: ",
                    "#C6C3C1",
                    ";
",
                  ],
                },
                "displayName": "styled.hr",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-iwsKbI",
                "target": "hr",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <hr
              className="sc-iwsKbI byAdSv"
            />
          </StyledComponent>
        </styled.hr>
        <Styled(styled.div)>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "fslGAT",
                  "rules": Array [
                    "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                    "
  flex-direction: row;
  justify-content: flex-start;
  padding: 0 1rem;
",
                  ],
                },
                "displayName": "Styled(styled.div)",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                ],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bdVaJa sc-htpNat fslGAT"
            >
              <styled.h3>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "cVKdGy",
                        "rules": Array [
                          "
  text-indent: 1rem;
",
                        ],
                      },
                      "displayName": "styled.h3",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "h3",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <h3
                    className="sc-bxivhb cVKdGy"
                  >
                    친구
                  </h3>
                </StyledComponent>
              </styled.h3>
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "inZLTK",
                        "rules": Array [
                          "
  text-indent: 0.5rem;
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-ifAKCX inZLTK"
                  >
                    2
                  </span>
                </StyledComponent>
              </styled.span>
            </div>
          </StyledComponent>
        </Styled(styled.div)>
        <UserCard
          key="234"
          onClick={[Function]}
          statusMessage="parkpark"
          userName="park"
        >
          <Styled(styled.div)
            onClick={[Function]}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-EHOje",
                    "isStatic": false,
                    "lastClassName": "VYauM",
                    "rules": Array [
                      "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                      "
  height: 5rem;
  justify-content: flex-start;
  padding: 0 1rem;
  background-color: ",
                      "#FEFFFF",
                      ";
  &:hover{
    background-color: ",
                      "#F1F2F2",
                      ";
  }
",
                    ],
                  },
                  "displayName": "Styled(styled.div)",
                  "foldedComponentIds": Array [
                    "sc-bdVaJa",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-EHOje",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <div
                className="sc-bdVaJa sc-EHOje VYauM"
                onClick={[Function]}
              >
                <Styled(styled.div)>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bZQynM",
                          "isStatic": false,
                          "lastClassName": "iEkADQ",
                          "rules": Array [
                            "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                            "
  justify-content: flex-start;

",
                          ],
                        },
                        "displayName": "Styled(styled.div)",
                        "foldedComponentIds": Array [
                          "sc-bdVaJa",
                        ],
                        "render": [Function],
                        "styledComponentId": "sc-bZQynM",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bdVaJa sc-bZQynM iEkADQ"
                    >
                      <TextIcon
                        color="#C6C3C1"
                        icon="Account"
                        text="park"
                      >
                        <styled.div
                          iconPosition="left"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-htoDjs",
                                  "isStatic": false,
                                  "lastClassName": "bdJNWD",
                                  "rules": Array [
                                    "
    display: flex;
    flex-direction: ",
                                    [Function],
                                    ";
    width: fit-content;
    height: fit-content;
    align-items: center;
    justify-content: center;
  ",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-htoDjs",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            iconPosition="left"
                          >
                            <div
                              className="sc-htoDjs bdJNWD"
                            >
                              <Icon
                                color="#C6C3C1"
                                icon="Account"
                              >
                                <ForwardRef(SvgAccount)
                                  fill="#C6C3C1"
                                  height="2rem"
                                  width="2rem"
                                >
                                  <svg
                                    fill="#C6C3C1"
                                    height="2rem"
                                    width="2rem"
                                  >
                                    account.svg
                                  </svg>
                                </ForwardRef(SvgAccount)>
                              </Icon>
                              <styled.span
                                text="park"
                                textColor="black"
                                textSize="1rem"
                              >
                                <StyledComponent
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-dnqmqq",
                                        "isStatic": false,
                                        "lastClassName": "eCurCS",
                                        "rules": Array [
                                          "
    color: ",
                                          [Function],
                                          ";
    line-height: 2rem;
    padding: 1rem;
  ",
                                        ],
                                      },
                                      "displayName": "styled.span",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "sc-dnqmqq",
                                      "target": "span",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  text="park"
                                  textColor="black"
                                  textSize="1rem"
                                >
                                  <span
                                    className="sc-dnqmqq eCurCS"
                                  >
                                    park
                                  </span>
                                </StyledComponent>
                              </styled.span>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </TextIcon>
                      <styled.span>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-gzVnrw",
                                "isStatic": false,
                                "lastClassName": "ielJtS",
                                "rules": Array [
                                  "
  font-size: ",
                                  "0.8rem",
                                  ";
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-gzVnrw",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="sc-gzVnrw ielJtS"
                          >
                            parkpark
                          </span>
                        </StyledComponent>
                      </styled.span>
                    </div>
                  </StyledComponent>
                </Styled(styled.div)>
              </div>
            </StyledComponent>
          </Styled(styled.div)>
        </UserCard>
        <UserCard
          key="345"
          onClick={[Function]}
          statusMessage="baebae"
          userName="bae"
        >
          <Styled(styled.div)
            onClick={[Function]}
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-EHOje",
                    "isStatic": false,
                    "lastClassName": "VYauM",
                    "rules": Array [
                      "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                      "
  height: 5rem;
  justify-content: flex-start;
  padding: 0 1rem;
  background-color: ",
                      "#FEFFFF",
                      ";
  &:hover{
    background-color: ",
                      "#F1F2F2",
                      ";
  }
",
                    ],
                  },
                  "displayName": "Styled(styled.div)",
                  "foldedComponentIds": Array [
                    "sc-bdVaJa",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-EHOje",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              onClick={[Function]}
            >
              <div
                className="sc-bdVaJa sc-EHOje VYauM"
                onClick={[Function]}
              >
                <Styled(styled.div)>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bZQynM",
                          "isStatic": false,
                          "lastClassName": "iEkADQ",
                          "rules": Array [
                            "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                            "
  justify-content: flex-start;

",
                          ],
                        },
                        "displayName": "Styled(styled.div)",
                        "foldedComponentIds": Array [
                          "sc-bdVaJa",
                        ],
                        "render": [Function],
                        "styledComponentId": "sc-bZQynM",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bdVaJa sc-bZQynM iEkADQ"
                    >
                      <TextIcon
                        color="#C6C3C1"
                        icon="Account"
                        text="bae"
                      >
                        <styled.div
                          iconPosition="left"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-htoDjs",
                                  "isStatic": false,
                                  "lastClassName": "bdJNWD",
                                  "rules": Array [
                                    "
    display: flex;
    flex-direction: ",
                                    [Function],
                                    ";
    width: fit-content;
    height: fit-content;
    align-items: center;
    justify-content: center;
  ",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-htoDjs",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            iconPosition="left"
                          >
                            <div
                              className="sc-htoDjs bdJNWD"
                            >
                              <Icon
                                color="#C6C3C1"
                                icon="Account"
                              >
                                <ForwardRef(SvgAccount)
                                  fill="#C6C3C1"
                                  height="2rem"
                                  width="2rem"
                                >
                                  <svg
                                    fill="#C6C3C1"
                                    height="2rem"
                                    width="2rem"
                                  >
                                    account.svg
                                  </svg>
                                </ForwardRef(SvgAccount)>
                              </Icon>
                              <styled.span
                                text="bae"
                                textColor="black"
                                textSize="1rem"
                              >
                                <StyledComponent
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-dnqmqq",
                                        "isStatic": false,
                                        "lastClassName": "eCurCS",
                                        "rules": Array [
                                          "
    color: ",
                                          [Function],
                                          ";
    line-height: 2rem;
    padding: 1rem;
  ",
                                        ],
                                      },
                                      "displayName": "styled.span",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "sc-dnqmqq",
                                      "target": "span",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  text="bae"
                                  textColor="black"
                                  textSize="1rem"
                                >
                                  <span
                                    className="sc-dnqmqq eCurCS"
                                  >
                                    bae
                                  </span>
                                </StyledComponent>
                              </styled.span>
                            </div>
                          </StyledComponent>
                        </styled.div>
                      </TextIcon>
                      <styled.span>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-gzVnrw",
                                "isStatic": false,
                                "lastClassName": "ielJtS",
                                "rules": Array [
                                  "
  font-size: ",
                                  "0.8rem",
                                  ";
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
",
                                ],
                              },
                              "displayName": "styled.span",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-gzVnrw",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <span
                            className="sc-gzVnrw ielJtS"
                          >
                            baebae
                          </span>
                        </StyledComponent>
                      </styled.span>
                    </div>
                  </StyledComponent>
                </Styled(styled.div)>
              </div>
            </StyledComponent>
          </Styled(styled.div)>
        </UserCard>
      </div>
    </StyledComponent>
  </Styled(styled.div)>
</UserList>
`;
